tests/boots/test_solved_software_environment.py

```diff
@@ -59,9 +61,13 @@ def test_run(self, context: Context) -> None:
         context.project.runtime_environment.operating_system.name = "fedora"
         context.project.runtime_environment.operating_system.version = "32"
         context.project.runtime_environment.python_version = "3.8"
-        context.graph.should_receive("solved_software_environment_exists").with_args(
-            os_name="fedora", os_version="32", python_version="3.8"
-        ).and_return(True).once()
+        assert {
+            "message": f"Consider using {context.project.runtime_environment.operating_system.name} "
+            f"in version {context.project.runtime_environment.operating_system.version} "
+            f"with Python {context.project.runtime_environment.python_version}",
+            "type": "ERROR",
+            "link": jl("solved_sw_env"),
+        } in context.stack_info
```

Comment:
You can do an exact match here to make sure there is always just one entry added.